bindings/python/CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+find_package(PythonInterp REQUIRED)
+find_package(PythonLibs REQUIRED)
+
+if (NOT boost-python)
+	# use active python
+	set(_boost_python "python${PYTHON_VERSION_MAJOR}")
```

Comment:
I don't know what to do. I'm on Gentoo and here we have python3.6m and boost_python-3.6, so the naming is up to package maintainers. I have no preference for the default name, the error message from CMake is pretty clear when the name is incorrect.

Reply:
There needs to be a special case for when `${PYTHON_VERSION_MAJOR} == 2`.

Reply:
I disagree. It seems simpler for me to pass -Dboost-python option to CMake than complicate the code here.

Reply:
Simpler for us sure, but not for people trying to build libtorrent. It'd be pretty silly to require explicitly setting boost python's library name even with a default boost install.

Reply:
But python2? Isn't it too old to provide a special handling for it? Perhaps there are more systems with non-default library names (like Gentoo) than those interested in bindings for python 2 only? Because those who need bindings for 3 and 2 have to read this fragment and the shorter it is the better.

Reply:
From the other hand, the current default does not make sense at all, I could just write "python3" instead...

Reply:
Arch:
```
libboost_python.a
libboost_python3.a
```
Fedora:
```
libboost_python.so.1.66.0
libboost_python3.so.1.66.0
```
Gentoo:
```
libboost_python-3.6-mt.so
libboost_python-3.6.a
libboost_python-3.6.so
```
Ubuntu:
```
libboost_python-py27.so.1.62.0
libboost_python-py35.so.1.62.0
libboost_python-py36.so.1.62.0
```

Reply:
Put it as a comment at the top of the file.

The default suffix should be `python` if the major version is 2 else `python${major}`. I'm not too worried about what all the different Linux distros do. They have package maintainers who can figure out what needs to be done to make it work with their unique naming schemes. I'm worried about Windows users. They all use the boost build naming, they're more likely to be building libtorrent manually, and they're the most likely to post a github issue when they get an error.

Reply:
Done